analyze/analyses/paths/mod.rs

```diff
@@ -0,0 +1,143 @@
+use std::collections::BTreeSet;
+
+use regex;
+
+use twiggy_ir as ir;
+use twiggy_opt as opt;
+use twiggy_traits as traits;
+
+mod paths_emit;
+mod paths_entry;
+
+use self::paths_entry::PathsEntry;
+
+#[derive(Debug)]
+struct Paths {
+    opts: opt::Paths,
+    entries: Vec<PathsEntry>,
+}
+
+/// Find all retaining paths for the given items.
+pub fn paths(items: &mut ir::Items, opts: &opt::Paths) -> Result<Box<traits::Emit>, traits::Error> {
+    // The predecessor tree only needs to be computed if we are ascending
+    // through the retaining paths.
+    if !opts.descending() {
+        items.compute_predecessors();
+    }
+
+    // Initialize the collection of Id values whose retaining paths we will emit.
+    let opts = opts.clone();
+    let starting_positions = get_items(&items, &opts)?;
+    let entries = starting_positions
+        .iter()
+        .map(|id| create_entry(*id, &items, &opts, &mut BTreeSet::new()))
+        .collect();
+
+    let paths = Paths {
+        opts,
+        entries,
+    };
+
+    Ok(Box::new(paths) as Box<traits::Emit>)
+}
+
+/// This helper function is used to collect ir::Id values for the `items` member
+/// of the `Paths` object, based on the given options.
+pub fn get_items(items: &ir::Items, opts: &opt::Paths) -> Result<Vec<ir::Id>, traits::Error> {
```

Comment:
Nit: This function should be renamed `get_starting_positions`, and have its documentation comment updated.